pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -309,34 +310,10 @@ func (h *mtuHandler) forceMssClamping(endpoint *submV1.Endpoint) error {
 		constants.LocalCIDRIPSet, "dst", "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j", "TCPMSS",
 		"--set-mss", strconv.Itoa(tcpMssValue),
 	}
+	rules = append(rules, ruleSpecSource, ruleSpecDest)
 
-	rules, err := h.ipt.List(constants.MangleTable, constants.SmPostRoutingChain)
-	if err != nil {
-		return errors.Wrapf(err, "error listing the rules in %s chain", constants.SmPostRoutingChain)
-	}
-
-	isPresent := false
-
-	for _, rule := range rules {
-		if strings.Contains(rule, strings.Join(ruleSpecSource, " ")) || strings.Contains(rule, strings.Join(ruleSpecDest, " ")) {
-			isPresent = true
-			break
-		}
-	}
-
-	if len(rules) > 0 && !isPresent {
-		if err := h.ipt.ClearChain(constants.MangleTable, constants.SmPostRoutingChain); err != nil {
-			logger.Warningf("Error flushing iptables chain %q of %q table: %v", constants.SmPostRoutingChain,
-				constants.MangleTable, err)
-		}
-	}
-
-	if err := h.ipt.AppendUnique(constants.MangleTable, constants.SmPostRoutingChain, ruleSpecSource...); err != nil {
-		return errors.Wrapf(err, "error appending iptables rule %q", strings.Join(ruleSpecSource, " "))
-	}
-
-	if err := h.ipt.AppendUnique(constants.MangleTable, constants.SmPostRoutingChain, ruleSpecDest...); err != nil {
-		return errors.Wrapf(err, "error appending iptables rule %q", strings.Join(ruleSpecDest, " "))
+	if err := h.ipt.UpdateChainRules(constants.MangleTable, constants.SmPostRoutingChain, rules); err != nil {
```

Comment:
```suggestion
	if err := h.ipt.UpdateChainRules(constants.MangleTable, constants.SmPostRoutingChain, [][]string{ruleSpecSource, ruleSpecDest}); err != nil {
```

and drop the references to `rules` above; or if you prefer to keep the separate variable,

```suggestion
	rules := [][]string{ruleSpecSource, ruleSpecDest}

	if err := h.ipt.UpdateChainRules(constants.MangleTable, constants.SmPostRoutingChain, rules); err != nil {
```

and drop the declaration at the start of the function.